apps/gateway/src/middlewares/models/index.ts

```diff
@@ -0,0 +1,61 @@
+import { t, type Static } from "elysia";
+
+import { CohereEmbedV4Adapter } from "./cohere";
+import {
+  Gemini25FlashPreviewAdapter,
+  Gemini25FlashLitePreviewAdapter,
+} from "./gemini";
+import { GptOss120bAdapter, GptOss20bAdapter } from "./gpt";
+
+import type { ModelAdapter } from "./model";
+
+export const supportedModel = t.Object({
+  id: t.String(),
+  name: t.String(),
+  created: t.Number(),
+  owned_by: t.String(),
+  modality: t.Union([t.Literal("chat"), t.Literal("embedding")]),
+  pricing: t.Object({
+    monthly_free_tokens: t.Number(),
+  }),
+});
+
+export type SupportedModel = Static<typeof supportedModel>;
+
+const MODEL_ADAPTER_MAP = {
+  "google/gemini-2.5-flash-preview-09-2025": () =>
+    new Gemini25FlashPreviewAdapter(),
+  "google/gemini-2.5-flash-lite-preview-09-2025": () =>
+    new Gemini25FlashLitePreviewAdapter(),
+  "openai/gpt-oss-120b": () => new GptOss120bAdapter(),
+  "openai/gpt-oss-20b": () => new GptOss20bAdapter(),
+  "cohere/embed-v4.0": () => new CohereEmbedV4Adapter(),
+};
+
+export type SupportedModelType = keyof typeof MODEL_ADAPTER_MAP;
+
+export const ModelAdapterFactory = {
+  getAdapter(modelType: SupportedModelType): ModelAdapter {
+    const factoryMethod = MODEL_ADAPTER_MAP[modelType];
+    if (!factoryMethod) {
+      throw new Error(`No model adapter found for model type: ${modelType}`);
+    }
+    return factoryMethod();
+  },
+};
+
+export function getSupportedModels(): SupportedModel[] {
+  return Object.values(MODEL_ADAPTER_MAP).map((factory) => {
+    const adapter = factory();
+    return {
```

Comment:
Any benefit of all these getters instead of the ModelAdapter returning an object of this shape?